XamarinCommunityToolkit/Views/DrawingView/Renderer/DrawingViewRenderer.macos.cs

```diff
@@ -0,0 +1,182 @@
+
+        void UpdatePath()
+        {
+            var smoothedPoints = Element.EnableSmoothedPath
+                ? SmoothedPathWithGranularity(points, Element.Granularity, ref currentPath)
+                : new List<Point>(points);
+            InvokeOnMainThread(Layer.SetNeedsDisplay);
+            points.Clear();
+            points.AddRange(smoothedPoints);
+        }
+
+        List<Point> SmoothedPathWithGranularity(List<Point> currentPoints,
+            int granularity,
+            ref NSBezierPath smoothedPath)
+        {
+            // not enough points to smooth effectively, so return the original path and points.
+            if (currentPoints.Count < 4)
+            {
+                return new List<Point>(currentPoints);
+            }
+
+            // create a new bezier path to hold the smoothed path.
+            smoothedPath.RemoveAllPoints();
+            var smoothedPoints = new List<Point>();
+
+            // duplicate the first and last points as control points.
+            currentPoints.Insert(0, currentPoints[0]);
+            currentPoints.Add(currentPoints[currentPoints.Count - 1]);
+
+            // add the first point
+            smoothedPath.MoveTo(currentPoints[0].X, currentPoints[0].Y);
+            smoothedPoints.Add(currentPoints[0]);
+
+            for (var index = 1; index < currentPoints.Count - 2; index++)
+            {
+                var p0 = currentPoints[index - 1];
+                var p1 = currentPoints[index];
+                var p2 = currentPoints[index + 1];
+                var p3 = currentPoints[index + 2];
+
+                // add n points starting at p1 + dx/dy up until p2 using Catmull-Rom splines
+                for (var i = 1; i < granularity; i++)
+                {
+                    var t = i * (1f / granularity);
+                    var tt = t * t;
+                    var ttt = tt * t;
+
+                    // intermediate point
+                    var mid = new Point
+                    {
+                        X = 0.5f *
+                            ((2f * p1.X) +
+                             ((p2.X - p0.X) * t) +
+                             (((((2f * p0.X) - (5f * p1.X)) + (4f * p2.X)) - p3.X) * tt) +
+                             ((((3f * p1.X) - p0.X - (3f * p2.X)) + p3.X) * ttt)),
+                        Y = 0.5f *
+                            ((2 * p1.Y) +
+                             ((p2.Y - p0.Y) * t) +
+                             (((((2 * p0.Y) - (5 * p1.Y)) + (4 * p2.Y)) - p3.Y) * tt) +
+                             ((((3 * p1.Y) - p0.Y - (3 * p2.Y)) + p3.Y) * ttt))
+                    };
+                    smoothedPath.LineTo(mid.X, mid.Y);
+                    smoothedPoints.Add(mid);
+                }
+
+                // add p2
+                smoothedPath.LineTo(p2.X, p2.Y);
+                smoothedPoints.Add(p2);
+            }
+
+            // add the last point
+            var last = currentPoints[currentPoints.Count - 1];
```

Comment:
As you used on iOS implementation (;
```suggestion
           var last = currentPoints[^1];
```